content/blog/2023/2023-05-03-react-and-spring-boot.md

```diff
@@ -0,0 +1,704 @@
+```javascript
+import React from "react";
+
+const PublisherList = ({ publishers, editEmployee, DeleteEmployee }) => {
+  return (
+    <table className="table table-hover mt-3" align="center">
+      <thead className="thead-light">
+        <tr>
+          <th scope="col">Nº</th>
+          <th scope="col">Name</th>
+          <th scope="col">Email</th>
+          <th scope="col">Published</th>
+          <th scope="col">Draft</th>
+
+          <th scope="col">Option</th>
+        </tr>
+      </thead>
+      {publishers.map(function fn(employee, index) {
+        return (
+          <tbody key={employee.id}>
+            <tr>
+              <th scope="row">{index + 1} </th>
+              <td>{employee.name}</td>
+              <td>{employee.email}</td>
+              <td>{employee.draft}</td>
+              <td>{employee.published}</td>
+              <td>
+                <button
+                  type="button"
+                  className="btn btn-warning"
+                  onClick={() => editEmployee(employee)}
+                >
+                  Edit
+                </button>
+                <button
+                  type="button"
+                  className="btn btn-danger mx-2"
+                  onClick={() => DeleteEmployee(employee.id)}
+                >
+                  Delete
+                </button>
+              </td>
+            </tr>
+          </tbody>
+        );
+      })}
+    </table>
+  );
+};
+
+export default PublisherList;
+```
+
+In the `PublisherList` component above, we are destructuring the passed props and using Bootstrap classes to display a table with the list of all saved publishers.
+
+Next in React, the `App.js` file is the root component of the application, and it is responsible for rendering and displaying all other components in the application. To display components in the application, we first need to import the components from their respective files, and then render them in the `App.js` file.
+
+To render our component, Paste the following code in our `App.js` file:
+
+```javascript
+import "bootstrap/dist/css/bootstrap.css";
+import "./App.css";
+import PublisherCrud from "./components/PublisherCrud";
+
+function App() {
+  return (
+    <>
+      <PublisherCrud />
+    </>
+  );
+}
+
+export default App;
+```
+
+Our application is now up and ready to communicate with our Spring Boot application.
+
+We can open the application on any browser of choice at `localhost:3000`.
+
+{{% image alt="spring-boot-and-reactjs-app" src="images/posts/spring-boot-and-reactjs/spring-boot-react-app.png" %}}
+
+## Conclusion
+By combining the strengths of Spring Boot and React, we can create a responsive, scalable, and modern web application. With careful planning, project structuring, and integration, we can provide a seamless user experience.
+
+You can refer to all the source code used in the article on [Github](https://github.com/thombergs/code-examples/tree/master/spring-boot/spring-boot-and-reactjs.
```

Comment:
The link format is broken

Reply:
Fixed. But my PR in the codeexample repo has to be merged yet